subsys/bluetooth/host/hci_core.c

```diff
@@ -6268,8 +7487,8 @@ static bool valid_le_scan_param(const struct bt_le_scan_param *param)
 		return false;
 	}
 
-	if (param->filter_dup &
-	    ~(BT_LE_SCAN_FILTER_DUPLICATE | BT_LE_SCAN_FILTER_WHITELIST)) {
+	if (param->options & ~(BT_LE_SCAN_OPT_FILTER_DUPLICATE |
+			       BT_LE_SCAN_OPT_FILTER_WHITELIST)) {
```

Comment:
I don't know what I should do, I've never participated in pull request discussion. But it seems another options checking missed.

Reply:
Yeah, that is right, it is missing.

Reply:
> Added this as draft PR to get input on the general directions. There are many small details that we can leave for later.

I forgot about it.